vscode/src/jdkDownloader.ts

```diff
@@ -173,8 +188,16 @@ export function JDKDownloader(JDKType: string, osType: string, osArchitecture: s
 
     writeStream.on('finish', async () => {
       vscode.window.showInformationMessage(`${JDKType} ${JDKVersion} for ${osType} download completed!`);
-
-      await extractJDK(path.join(targetDirectory, fileName), installationPath, JDKVersion, osType);
+      const checkSumObtained = calculateChecksum(filePath);
+      const checkSumExpected = (await axios.get(`${downloadUrl}.sha256`)).data;
+      
+      if (checkSumExpected === checkSumObtained) {
+        await extractJDK(filePath, installationPath, JDKVersion, osType);
+      }
+      else {
+        handleLog(logger, "Checksums don't match!");
+        throw new Error("Something went wrong");
```

Comment:
I tried what happens when the checksum fails, by hardcoding:
```
handleLog(logger, "Checksums don't match!");
throw new Error("Something went wrong");
```
after `showInformationMessage`. And it does not seem very user-friendly to me: it will show "... download completed!", and that's it. I think I would move the "... download completed!" message to the branch where the checksum passes, and would use something like `showErrorMessage` to report the problem. Possibly something along the lines of "${JDKType} ${JDKVersion} for {$osType} download failed, wrong checksum."

The `throws new Error` also seems a bit harsh - when the message is show, maybe we could simply not do anything else?

Reply:
Yes, it didn't seem user friendly, I have updated the code according to the comment given by you. Thanks.